src/yoroi-wallets/cardano/transactionManager.ts

```diff
@@ -0,0 +1,34 @@
+import {Storage} from '../storage'
+import {BackendConfig} from '../types'
+import {TransactionCache} from './shelley'
+
+export const makeTransactionManager = async (storage: Storage) => {
+  const memosStorage = storage.join('memos/')
+  const transactionCache = await TransactionCache.create(storage.join('txs/'), memosStorage)
+
+  return {
+    // transactionCache api
+    get transactions() {
+      return transactionCache.transactions
```

Comment:
I'd remove all changes in the tx cache, the memo is wallet info, and is just linked to tx, why don't modify only the getter to combine the data, when present, here?

Reply:
Getting the memos is an async operation, that's the reason I'm adding in advance the memos.

Do you have any suggestion?

Reply:
I can convert the getter to asynchronous but then I would have to make modifications in the parts where it is being consumed. If you have an different idea please let me know

Reply:
you could load all the memos into memory and merge on the fly

Reply:
or make a new wallet method `await wallet.getTransaction(txid)` that wraps `wallet.transactionManager.getTransaction(txid)`

Reply:
it looks like `getTransactionInfos` and `wallet.getTransactions` do basically the same things, could use some cleanup

Reply:
> you could load all the memos into memory and merge on the fly

@wolverineks At what point in the wallet cycle would you load it?

> or make a new wallet method await wallet.getTransaction(txid) that wraps wallet.transactionManager.getTransaction(txid)

when would that feature be useful? could you extend this idea? Thanks

Reply:
Maybe one idea is to create a method to load the memos and subscribe to `transactionCache` `transaction` event

Reply:
make transaction manager?

Reply:
no event subscriptions
load them into memory when creating the transaction manager